gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineCheckstyle.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * (c) Copyright 2015 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.plugins;
+
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+import org.gradle.api.Project;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.plugins.quality.Checkstyle;
+import org.gradle.api.plugins.quality.CheckstyleExtension;
+import org.gradle.api.plugins.quality.CheckstylePlugin;
+import org.gradle.api.tasks.javadoc.Javadoc;
+import org.gradle.external.javadoc.StandardJavadocDocletOptions;
+import org.gradle.plugins.ide.eclipse.model.EclipseModel;
+import org.gradle.plugins.ide.eclipse.model.EclipseProject;
+
+/**
+ * Configures the Gradle "checkstyle" task with Baseline settings.
+ */
+public final class BaselineCheckstyle extends AbstractBaselinePlugin {
+
+    private static final String DEFAULT_CHECKSTYLE_VERSION = "8.11";
```

Comment:
I guess we should probably cut and RC so that we can try out the 7.5.1 -> 8.11 upgrade on a few internal projects?

Reply:
want to go straight to 8.12?